pkg/api/api.go

```diff
@@ -224,3 +228,139 @@ type Message struct {
 	MessageID ulid.ULID   `json:"message_id"`
 	Payload   []byte      `json:"payload,omitempty"`
 }
+
+type Validator interface {
+	// checks whether this field is one of the valid options
+	Validate() error
+}
+
+// checks every field of the state, returning an error when any invalid field is encountered
+func (state State) Validate() error {
+
+	for _, field := range []Validator{
+		state.VisionStatus,
+		state.BallFound,
+		state.LocalizationStatus,
+		state.CPB,
+		state.Role,
+		state.RefBoxRole,
+		state.HomeGoal,
+		state.TeamColor,
+		state.CapacitorState,
+		state.Kinect1State,
+		state.Kinect2State,
+	} {
+		// the String() method is required, and also works for non-string Validators
+		if reflect.ValueOf(field).String() != reflect.Zero(reflect.TypeOf(field)).String() {
+			if err := field.Validate(); err != nil {
+				return err
+			}
+		}
+	}
+
+	switch {
+	case state.CamStatus > 10:
+		return rangeError("Cam status")
+	case state.RestartCountVision > 99:
+		return rangeError("Restart counter of Vision Executable")
+	case state.RestartCountWorldmodel > 99:
+		return rangeError("Restart counter of World Model")
+	case state.BatteryVoltage > 99:
+		return rangeError("Battery voltage")
+	case state.EmergencyStatus > 100:
+		return rangeError("Emergency status")
+	case state.Cpu0Load > 99:
+		return rangeError("Cpu 0 load")
+	case state.Cpu1Load > 99:
+		return rangeError("Cpu 1 load")
+	case state.ActiveDevPC > 90:
+		return rangeError("Active dev pc")
+	case state.Temperature1 > 99:
+		return rangeError("Temperature 1")
+	case state.Temperature2 > 99:
+		return rangeError("Temperature 2")
+	case state.Temperature3 > 99:
+		return rangeError("Temperature 3")
+	}
+	return nil
+}
+
+// return an error
+func rangeError(source string) error {
+	return errors.New(source + " out of range")
+}
+
+func (item VisionStatus) Validate() error {
+	switch item {
+	case VisionStatusManual, VisionStatusOn, VisionStatusOff:
+	default:
+		return errors.New("Invalid VisionStatus: got '" + string(item) + "'")
+	}
+	return nil
+}
+
+func (item BallFound) Validate() error {
+	if !(item == BallFoundYes || item == BallFoundNo || item == BallFoundCommunicatied) {
+		return errors.New("Invalid BallFound: got '" + string(item) + "'")
+	}
+	return nil
+}
+
+func (item LocalizationStatus) Validate() error {
+	if !(item == LocalizationStatusOn || item == LocalizationStatusOff || item == LocalizationStatusManual) {
+		return errors.New("Invalid Localisation status: got '" + string(item) + "'")
+	}
+	return nil
+}
+
+func (item CPB) Validate() error {
+	if !(item == CPBNo || item == CPBYes || item == CPBCommunicatied) {
```

Comment:
Use `switch` statements in these `Validate()` methods.

Reply:
we discussed this. I can do that, but it isn't necessary

Reply:
This is more shorter, so I prefer I you leave it this way

Reply:
Go is a verbose language.

Reply:
_dude_
for this case: imo your switch case is only readable for one who have seen it before.
the if-checks are unambiguous: these do not depend on a very specific go-thing
"it is go" != "it is readable"

Reply:
People not knowing what a `switch` statement is should not be a reason for not using it. We might as well not use pointers, maps and channels because some of us might be unfamiliar with the concept.

It's fine to use an `if` here as long as you use it everywhere else also.
The problem here is that you didn't reply to the comment, which means you agree and are in-progress of fixing it.